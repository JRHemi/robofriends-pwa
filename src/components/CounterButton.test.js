import React from "react";
import { shallow } from "enzyme";
import CounterButton from "./CounterButton";
import { count } from "console";

describe("CounterButton Component", () => {
  // Ensure mocks are cleared after each test
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("expects to render CounterButton component", () => {
    const mockColor = "green";
    expect(shallow(<CounterButton color={mockColor} />)).toMatchSnapshot();
  });

  it("expects click to increment the counter", () => {
    const mockColor = "red";
    const wrapper = shallow(<CounterButton color={mockColor} />);

    expect(wrapper.state()).toEqual({ count: 0 });
    wrapper.find('[id="counter"]').simulate("click");
    expect(wrapper.state()).toEqual({ count: 1 });
    expect(wrapper.props().color).toEqual("red");
  });

  it("expects component will only update when count changes", () => {
    const mockColor = "red";
    const wrapper = shallow(<CounterButton color={mockColor} />);
    expect(wrapper.state()).toEqual({ count: 0 });

    const instance = wrapper.instance();
    expect(instance.shouldComponentUpdate({ count: 0 })).toEqual(false);
    expect(instance.shouldComponentUpdate({ count: 1 })).toEqual(true);
  });
});
