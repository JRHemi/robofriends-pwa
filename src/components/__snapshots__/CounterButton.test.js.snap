// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton Component expects to render CounterButton component 1`] = `
<button
  color="green"
  id="counter"
  onClick={[Function]}
  style={
    Object {
      "color": "green",
    }
  }
>
  Count: 
  0
</button>
`;
